src/reconnectcontroller/DefaultReconnectController.ts

```diff
@@ -25,24 +30,47 @@ export default class DefaultReconnectController implements ReconnectController {
     return Date.now() - this.firstConnectionAttemptTimestamp;
   }
 
+  private timeSinceLastActiveMs(): number {
+    if (!this.canReceivePongs) {
+      return 0;
+    }
+    return Date.now() - this.lastConnectionAttemptTimestamp;
+  }
+
   private hasPastReconnectDeadline(): boolean {
-    return this.timeSpentReconnectingMs() >= this.reconnectTimeoutMs;
+    const hasSpentTooMuchTimeReconnecting =
+      this.timeSpentReconnectingMs() >= this.reconnectTimeoutMs;
+    const hasBeenTooLongSinceConnected = this.timeSinceLastActiveMs() >= this.reconnectTimeoutMs;
+    return hasSpentTooMuchTimeReconnecting || hasBeenTooLongSinceConnected;
+  }
+
+  didReceivePong(_latencyMs: number, _clockSkewMs: number): void {
+    this.lastConnectionAttemptTimestamp = Date.now();
+    this.canReceivePongs = true;
   }
 
   reset(): void {
     this.cancel();
+    if (this.firstConnectionAttempted || this.canReceivePongs) {
+      this.pingPong.removeObserver(this);
+      this.pingPong.stop();
+      this.canReceivePongs = false;
+    }
     this.shouldReconnect = true;
     this.onlyRestartPeerConnection = false;
     this.firstConnectionAttempted = false;
     this.firstConnectionAttemptTimestamp = 0;
     this.backoffPolicy.reset();
   }
 
-  startedConnectionAttempt(isFirstConnection: boolean): void {
+  startedConnectionAttempt(isFirstConnection: boolean, pingPong: PingPong): void {
     this._isFirstConnection = isFirstConnection;
     if (!this.firstConnectionAttempted) {
       this.firstConnectionAttempted = true;
       this.firstConnectionAttemptTimestamp = Date.now();
+      this.pingPong = pingPong;
+      this.pingPong.addObserver(this);
+      this.pingPong.start();
```

Comment:
The `PingPong` is already started and stopped in the connection monitor: https://github.com/aws/amazon-chime-sdk-js/blob/master/src/connectionmonitor/SignalingAndMetricsConnectionMonitor.ts#L43

Why do we need to start and stop it here? We should avoid having two different objects controlling the lifecycle of `PingPong`.